Kian-Aghaei/wisdom
# Adding animation to the search box

**What is done:**
- edition to Search.js component
- Adding required CSS to the main.css

**How to test:**
- click on the magnifying glass icon
- the text in front of it must disappear (SEARCH)
- The icon should turn into a box for typing the search keywords
- A close sign must appear at the rightmost side of the box
- Clicking on the close sign must close the box and reappear the text in front of it (SEARCH)

**Snapshot:**

![S1](https://user-images.githubusercontent.com/37697445/59359020-e9f98780-8d2d-11e9-96cb-6a8e6834fdc4.jpg)
![S2](https://user-images.githubusercontent.com/37697445/59359034-ee25a500-8d2d-11e9-8064-5b308d6f4d8c.jpg)

## Update: 14/06/2019

**What is done:**
- Adding the functionality to close the primary form of the search box by pressing the Esc key
- Modifying the functionality of the search box regarding screen width
- Search box only expand on its own in the default mode while screen width is above 800px
- Adding a search Box while window width is less than 800px
- Configuring both primary and secondary form of search boxes to be fully responsive 

**How to test:**
- Pressing the Esc key should close the opened search box (the only primary form of it)
- By clicking on the magnifying glass icon while the screen width is above 800px, you should see the default behavior which is search box icon (magnifying glass) turns into a box (input) and again it should be back to its primary shape.
-  By clicking on the magnifying glass icon while the screen width is below 800px, you should see a search box appears underneath the logo and by clicking again it must disappear.

![mSearch1](https://user-images.githubusercontent.com/37697445/59515988-a7b47f80-8ec0-11e9-8100-85b3314bfdd6.png)
![mSearch2](https://user-images.githubusercontent.com/37697445/59515994-ac793380-8ec0-11e9-8b9c-3a85175ca080.png)

## Update: 15/06/2019

- Fixing the issue of the search box and the logo collision in the header while the screen width is less than 800px.
- Fixing some of the issues in the comments and the behavior of the secondary search box
